wallet-core/src/ffi.rs

```diff
@@ -61,16 +106,59 @@ pub unsafe extern "C" fn generate_profile(
         BlsPublicKey::SIZE,
     );
 
-    0
+    ErrorCode::Ok
 }
 
-// Currently we're not handling panic message in the WASM module; in the future
-// we might want to enable it for `debug` releases.
-mod panic_handling {
-    use core::panic::PanicInfo;
+/// Filter all notes and their block height that are owned by the given keys,
+/// mapped to their nullifiers.
+#[no_mangle]
+pub unsafe fn map_owned(
+    seed: &Seed,
+    indexes: *const u8,
+    notes_ptr: *mut u8,
+) -> ErrorCode {
+    let keys = indexes_into_keys(seed, indexes, derive_phoenix_sk);
+    let notes = read_buffer(notes_ptr);
+    let notes: Vec<NoteLeaf> = from_bytes::<Vec<NoteLeaf>>(&notes)
+        .or(Err(ErrorCode::UnarchivingError))?;
 
-    #[panic_handler]
-    fn panic(_info: &PanicInfo) -> ! {
-        loop {}
-    }
+    let owned = notes::map_owned(&keys, notes);
+
+    keys.into_iter().for_each(|mut sk| sk.zeroize());
+
+    let bytes =
+        to_bytes::<_, 4096>(&owned).or(Err(ErrorCode::ArchivingError))?;
```

Comment:
Can we use `NoteLeaf::SIZE` instead of `4096`?

Reply:
We can't, this is a buffer; it means even if it holds a multiple of `NoteLeaf::SIZE` it will still need to handle additional metadata. Plus, we don't really knows how many `NoteLeaf` the host is sending us; depends on the sync process and the settings of the application: it could be 100 notes as well as a couple or 58Mb of notes.

But thanks for pointed this out, since the 4k was meant only for the testing; I'm actually going to put a proper scratch buffer size most likely used for the sync.